### PR TITLE
Health Insurance Cross-Sell Prediction

### PR DESCRIPTION
**Pull request by siddhant4ds**
Closes:  #74

- **Title** : Health Insurance Cross-Sell Prediction
- **Name:**  Siddhant Tiwari
- **Idenitfy yourself:**  SSOC-3 Contributor

### Describe the add-ons or changes you've made 📃

Linked a notebook which includes:
1. Exploratory Data Analysis of the dataset
2. Data cleaning and preprocessing as required for different ML algorithms
3. Feature engineering
4. Feature selection based on statistical tests and feature-importances.
5. Models implemented: Logistic Regression, Linear Discriminant Analysis, Quadratic Discriminant Analysis, Gaussian Naive Bayes, K-Nearest Neighbors, Decision Tree, Random Forest, AdaBoost, Multi-Layer Perceptron neural network, XGBoost, CatBoost and LightGBM
6. Post-tuning of prediction thresholds for models for more accurate predictions.
7. Result analysis using classification report along with ROC-AUC and Matthew's correlation coefficient.

Added a README which includes the project details and images generated.

### Type of change ☑️

What sort of change have you made:

- [x] New feature (non-breaking change which adds functionality)

### How Has This Been Tested? ⚙️

Reproducibility: Fixed random seed for all experiments and multiple runs to check consistency.
Dependencies: Reran notebook on local environment as well as cloud-hosted notebook (Kaggle).

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

scr1
scr2
scr3
scr4
scr5
scr6

### CODE REVIEW

**Comment by siddhant4ds** [comment]
@Avdhesh-Varshney Requesting a review. Thanks.

**Review by Avdhesh-Varshney — APPROVED** [review]
